Pull request team one

mryba opened the pull request:
pull request - team number one.

Commit:
Commit to rebase

smalaca commented on src/main/java/com/smalaca/messagesender/controller/MessageSenderController.java:
```diff
@@ -0,0 +1,34 @@
+    public MessageSenderController(
+            MessageSenderService messageSenderService, MessageCrud messageCrud, MessageRepository messageRepository) {
+        this.messageSenderService = messageSenderService;
+        this.messageCrud = messageCrud;
```
nie potrzebne

smalaca replied:
nie potrzebne

smalaca commented on src/main/java/com/smalaca/messagesender/controller/MessageSenderController.java:
```diff
@@ -0,0 +1,34 @@
+
+    @RequestMapping(value = "/send/{param}")
+    public String sendMessage(@PathVariable(value = "param") String id){
+        Response response1 =  messageSenderService.sendMessageViaEmail(id);
```
A może?
```java
return messageSenderService.sendMessageViaEmail(id);
```

smalaca replied:
Jeżeli nie to response jest nie potrzebny

smalaca commented on src/main/java/com/smalaca/messagesender/exceptions/NoMessageException.java:
```diff
@@ -0,0 +1,15 @@
+        this.messageId = messageId;
+    }
+
+    public String getMessageId() {
```
jeżeli nigdzie nie wykorzystujecie tej metody to ją usuńcie. To implikuje również usunięcie atrybutu 😄

smalaca commented on src/main/java/com/smalaca/messagesender/repository/inmemory/InMemoryMessageRepository.java:
```diff
@@ -19,20 +19,32 @@ public InMemoryMessageRepository(List<Message> messages) {
+        return messages.stream()
+                .filter(message -> message.hasSameId(id))
+                .findAny()
+                .get();
```
w tym momencie dostaniecie null pointer exception jeżeli message nie istnieje

smalaca commented on src/main/java/com/smalaca/messagesender/service/MailSender.java:
```diff
@@ -0,0 +1,19 @@
+import java.util.Properties;
+
+@Component
+public class MailSender {
```
to powinien być interfejs, a implementacja np. ```FakeMailSender```.

smalaca commented on src/test/java/com/smalaca/messagesender/controller/MessageSenderControllerTest.java:
```diff
@@ -0,0 +1,59 @@
+        messageDto.setFrom("msiek");
+        messageDto.setTo("java5krk");
+
+//        messageCrud.createNew(messageDto);
```
czegoś brakuje 👅

smalaca commented on src/test/java/com/smalaca/messagesender/repository/inmemory/InMemoryMessageRepositoryTest.java:
```diff
@@ -88,6 +91,19 @@ public void shouldRecognizeThatMessageSimilarToThirdDoesNotExist() {
+    public void shouldGetMessageByID() {
+        MessageDto messageDto = new MessageDto();
+        repository.add(new MessageFactory().createFrom(messageDto, "1"));
+        repository.getMessageById("1");
```
Dlaczego tyle razy getMessageId() ? Chyba rebase nie zadziałał zbyt dobrze 😝 

![image](https://user-images.githubusercontent.com/4753177/36946356-fa8934be-1fbb-11e8-8ff6-902789f6e20c.png)

smalaca commented on src/test/java/com/smalaca/messagesender/service/MailSenderTest.java:
```diff
@@ -0,0 +1,45 @@
+        Message message = messageRepository.getMessageById("1");
+        Assert.assertTrue(mailSender.sendEmailSender(message).isSuccess());
+    }
+
```
usuńcie niepotrzebne nowe linie

smalaca commented on src/test/java/com/smalaca/messagesender/service/MessageSenderServiceTest.java:
```diff
@@ -0,0 +1,52 @@
+
+    @Test
+    public void shouldReturnExceptionIfSendMessageViaMailUnsuccessful() {
+        Assert.assertTrue(true);
```
![image](https://user-images.githubusercontent.com/4753177/36946379-55542eee-1fbc-11e8-8682-b869d03094aa.png)

smalaca commented on src/test/java/com/smalaca/messagesender/service/MessageSenderServiceTest.java:
```diff
@@ -0,0 +1,52 @@
+            messageSenderService.sendMessageViaEmail(messageId);
+            fail("Expected NoMessageException");
+        } catch (NoMessageException e) {
+            e.printStackTrace();
```
assert na exception message.

Commit:
- controller/MessageSenderController.java - fix issue.
- controller/MessageSenderController.java - change sendMessage method.
- exceptions/NoMessageException.java - remove not uses getMessageId method
- inmemory/InMemoryMessageRepository.java - add exception to filtr stream.
 /service/MailSender.java - change MailSender class to Interface, and create FakeMailSender to implement MailSender interface.
 -InMemoryMessageRepositoryTest.java - clean up shouldGetMessageById method.
 -service/MailSenderTest.java - refactor code from blank space.
 - MessageSenderServiceTest.java - true - true ;X
 -MessageSenderServiceTest.java - add test

Commit:
MessageSenderControllerTest fix / update
